app/Services/BigCommerceService.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Config;
+use App\Models\Script;
+use Exception;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Session;
+use Psr\Http\Message\ResponseInterface;
+
+class BigCommerceService
+{
+    private Client $httpClient;
+
+    public function __construct(Client $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    public function verifySignedRequest(string $signedPayload): ?array
+    {
+        if (strpos($signedPayload, '.') !== false) {
+
+            list($encodedData, $encodedSignature) = explode('.', $signedPayload, 2);
+
+            $signature = base64_decode($encodedSignature);
+            $jsonStr = base64_decode($encodedData);
+            $data = json_decode($jsonStr, true);
+            $this->validateSignature($signature, $jsonStr);
+
+            if (!isset($data['owner']) || !isset($data['user']) || !isset($data['context'])) {
+                throw new Exception('The signed request from BigCommerce has missing data!');
+            }
+
+            return $data;
+        } else {
+            return null;
+        }
+    }
+
+    public function validateSignature($signature, $jsonStr) {
+        $expectedSignature = hash_hmac('sha256', $jsonStr, env('CLIENT_SECRET'), $raw = false);
```

Comment:
You're lucky here, because variable assignments somehow return `false`, but you're setting `$raw = false` as an argument. Any other language would have immediately screamed at you, but PHP seems to be fine with it.

```
php > var_dump($raw = false);
php shell code:1:
bool(false)
php > 
```

Reply:
:smile: someone should probably tell them. That example also contains some odd indents.

Reply:
Changed the code but BigCommerce need to take care about their documentation 😉